.circleci/config.yml

```diff
@@ -128,49 +131,61 @@ jobs:
             fi
       - save_cache: *save_cache
 
-  build-integration-cache:
+  build.fullstack.cache:
     docker:
       - <<: *base
     steps:
       - checkout
-      - run: *chown_hack_for_cache_integration
+      - run: *chown_hack_for_cache_fullstack
       - restore_cache: *restore_cache
-      - restore_cache: *restore_integration_cache
+      - restore_cache: *restore_fullstack_cache
       - run: find /var/cache/zypp/packages/ | wc -l
       - run:
           command: |
             ls -lRa /var/cache/zypp/packages/openQA || :
             if [ ! -d /var/cache/autoinst/t ]; then
               bash -x .circleci/build_autoinst.sh "/var/cache/autoinst"
             fi
-      - save_cache: *save_integration_cache
+      - save_cache: *save_fullstack_cache
 
-  test1:
+  t: &test-template
     docker:
       - <<: *base
+    environment:
+      TEST_PART=t
     steps:
       - checkout
       - run: *chown_hack_for_cache
       - restore_cache: *restore_cache
       - run: *check_cache
       - run: *install_cached_packages
-      - run: sudo zypper -n install python3-six # workaround bug https://bugzilla.suse.com/show_bug.cgi?id=1150895
-      - run: eval "$(t/test_postgresql | grep TEST_PG=)" && PERL5LIB=lib prove -v t/*.t
+      - run: *make_test
+      - persist_to_workspace:
+          root: .
+          paths:
+            - cover_db_*
 
-  testui:
+  t.fullstack: &test-fullstack-template
     docker:
       - <<: *base
+    environment:
+      TEST_PART=fullstack
```

Comment:
I like the direction you're heading, making the jobs more generic. I wonder, could we go one step further here and use `$CIRCLE_JOB` ie. select the test based on the name? Which should make those jobs one-liners.